packages/shell-api/src/helpers.ts

```diff
@@ -120,3 +122,322 @@ export function processDigestPassword(username, passwordDigestor, command): any
+    // Mongoses older than the threshold are the most recent, but cannot be
+    // considered "active" mongoses. (This is more likely to be an old(er)
+    // configdb dump, or all the mongoses have been stopped.)
+    if (mostRecentMongosTime.getTime() >= Date.now() - mongosActiveThresholdMs) {
+      mongosAdjective = 'active';
+    }
+  }
+
+  mongosAdjective = `${mongosAdjective} mongoses`;
+  if (mostRecentMongosTime === null) {
+    result[mongosAdjective] = 'none';
+  } else {
+    const recentMongosQuery = {
+      ping: {
+        $gt: ((): any => {
+          const d = mostRecentMongosTime;
+          d.setTime(d.getTime() - mongosActiveThresholdMs);
+          return d;
+        })()
+      }
+    };
+
+    if (verbose) {
+      result[mongosAdjective] = await mongosColl
+        .find(recentMongosQuery)
+        .sort({ ping: -1 })
+        .toArray();
+    } else {
+      result[mongosAdjective] = (await (await mongosColl.aggregate([
+        { $match: recentMongosQuery },
+        { $group: { _id: '$mongoVersion', num: { $sum: 1 } } },
+        { $sort: { num: -1 } }
+      ])).toArray()).map((z) => {
+        const res = {};
+        res[z._id] = z.num;
+        return res;
+      });
+    }
+  }
+
+  // Is autosplit currently enabled
+  const autosplit = await settingsColl.findOne({ _id: 'autosplit' }) as any;
+  result.autosplit = { 'Currently enabled': autosplit === null || autosplit.enabled ? 'yes' : 'no' };
+
+  // Is the balancer currently enabled
+  const balancerRes = {};
+  const balancerEnabled = await settingsColl.findOne({ _id: 'balancer' }) as any;
+  balancerRes['Currently enabled'] = balancerEnabled === null || !balancerEnabled.stopped ? 'yes' : 'no';
+
+  // Is the balancer currently active
+  let balancerRunning = 'unknown';
+  try {
+    const balancerStatus = await configDB.adminCommand({ balancerStatus: 1 });
+    balancerRunning = balancerStatus.inBalancerRound ? 'yes' : 'no';
+  } catch (err) {
+    // pass, ignore all error messages
+  }
+  balancerRes['Currently running'] = balancerRunning;
+
+  // Output the balancer window
+  const settings = await settingsColl.findOne({ _id: 'balancer' });
+  if (settings !== null && settings.hasOwnProperty('activeWindow')) {
+    const balSettings = settings.activeWindow;
+    balancerRes['Balancer active window is set between'] = `${balSettings.start} and ${balSettings.stop} server local time`;
+  }
+
+  // Output the list of active migrations
+  const activeLocks = await configDB.getCollection('locks').find({ state: { $eq: 2 } }).toArray();
+  const activeMigrations = [];
+  if (activeLocks !== null) {
+    activeLocks.forEach((lock) => {
+      activeMigrations.push({ _id: lock._id, when: lock.when });
+    });
+  }
+
+  if (activeMigrations.length > 0) {
+    balancerRes['Collections with active migrations'] = activeMigrations.map((migration) => {
+      return `${migration._id} started at ${migration.when}`;
+    });
+  }
```

Comment:
A lot of the db queries here are run sequentially, e.g. those that fill `balancerRes`, could we run them in parallel to reduce runtime?

Reply:
Sure, what would that look like? I haven't actually done that before :D

Reply:
So … I haven’t tested this, but I the general idea would be to turn e.g.

```js
// Is the balancer currently enabled
const balancerRes = {};
const balancerEnabled = await settingsColl.findOne({ _id: 'balancer' }) as any;
balancerRes['Currently enabled'] = balancerEnabled === null || !balancerEnabled.stopped ? 'yes' : 'no';

// Is the balancer currently active
let balancerRunning = 'unknown';
try {
  const balancerStatus = await configDB.adminCommand({ balancerStatus: 1 });
  balancerRunning = balancerStatus.inBalancerRound ? 'yes' : 'no';
} catch (err) {
  // pass, ignore all error messages
}
balancerRes['Currently running'] = balancerRunning;

// Output the balancer window
const settings = await settingsColl.findOne({ _id: 'balancer' });
if (settings !== null && settings.hasOwnProperty('activeWindow')) {
  const balSettings = settings.activeWindow;
  balancerRes['Balancer active window is set between'] = `${balSettings.start} and ${balSettings.stop} server local time`;
}

// Output the list of active migrations
const activeLocks = await configDB.getCollection('locks').find({ state: { $eq: 2 } }).toArray();
const activeMigrations = [];
if (activeLocks !== null) {
  activeLocks.forEach((lock) => {
    activeMigrations.push({ _id: lock._id, when: lock.when });
  });
}

if (activeMigrations.length > 0) {
  balancerRes['Collections with active migrations'] = activeMigrations.map((migration) => {
    return `${migration._id} started at ${migration.when}`;
  });
}
```

into something like

```js
// Is the balancer currently enabled
const balancerRes = {};

await Promise.all([
  (async () => {
    const balancerEnabled = await settingsColl.findOne({ _id: 'balancer' }) as any;
    balancerRes['Currently enabled'] = balancerEnabled === null || !balancerEnabled.stopped ? 'yes' : 'no';
  })(),
  
  (async () => {
    // Is the balancer currently active
    let balancerRunning = 'unknown';
    try {
      const balancerStatus = await configDB.adminCommand({ balancerStatus: 1 });
      balancerRunning = balancerStatus.inBalancerRound ? 'yes' : 'no';
    } catch (err) {
      // pass, ignore all error messages
    }
    balancerRes['Currently running'] = balancerRunning;
  })(),
  
  (async () => {
    // Output the balancer window
    const settings = await settingsColl.findOne({ _id: 'balancer' });
    if (settings !== null && settings.hasOwnProperty('activeWindow')) {
      const balSettings = settings.activeWindow;
      balancerRes['Balancer active window is set between'] = `${balSettings.start} and ${balSettings.stop} server local time`;
    }
  })(),

  (async () => {
    // Output the list of active migrations
    const activeLocks = await configDB.getCollection('locks').find({ state: { $eq: 2 } }).toArray();
    const activeMigrations = [];
    if (activeLocks !== null) {
      activeLocks.forEach((lock) => {
        activeMigrations.push({ _id: lock._id, when: lock.when });
      });
    }

    if (activeMigrations.length > 0) {
      balancerRes['Collections with active migrations'] = activeMigrations.map((migration) => {
        return `${migration._id} started at ${migration.when}`;
      });
    }
  })()
]);
```

This is probably not the only part here where we could do this, and maybe not ideal, but I think the idea is visible?

Reply:
yes, that makes sense, thanks!

Reply:
Ah I didn't push the change before merging, I will make a new PR